src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,65 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { FilterStatusType } from '../../types/Todo';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  setFilterLetter: (letter: string) => void;
+  setFilterStatus: (status: FilterStatusType) => void;
+  filterLetter: string;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  setFilterLetter,
+  setFilterStatus,
+  filterLetter,
+}) => {
+  const handleChangeFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFilterLetter(event.target.value);
+  };
+
+  const handleSelectFilter = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setFilterStatus(event.target.value as FilterStatusType);
+  };
+
+  const handlClearButton = () => {
+    setFilterLetter('');
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleSelectFilter}>
+            <option value={FilterStatusType.All}>All</option>
+            <option value={FilterStatusType.Active}>Active</option>
+            <option value={FilterStatusType.Completed}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          onChange={handleChangeFilter}
+          value={filterLetter}
+        />
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          {filterLetter !== '' && (
```

Comment:
- Avoid inline styles, use classes instead
- Don't disable linter
- And you don't need to compare with an empty string, just check if a value exists https://developer.mozilla.org/en-US/docs/Glossary/Truthy